extension/src/pipeline/quickPick.ts

```diff
@@ -79,36 +92,79 @@ const getFieldOptions = (data: UnknownValue): string[] => {
     : []
 }
 
+const showNotEnoughKeysToast = () =>
+  Toast.showError(
+    'The requested file(s) does not contain enough keys (columns) to generate a plot. Does the file or files follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+  )
+
+const getFieldsFromDataFiles = (
+  dataArr: { data: UnknownValue; file: string }[]
+) => {
+  const keys: {
+    [file: string]: string[]
+  } = {}
+  let keysAmount = 0
+
+  for (const { file, data } of dataArr) {
+    const fields = getFieldsFromValue(data)
+
+    if (fields.length === 0) {
+      void showNotEnoughKeysToast()
+      return
+    }
+
+    keysAmount += fields.length
+
+    if (!keys[file]) {
+      keys[file] = []
+    }
+    keys[file].push(...fields)
+  }
+
+  if (keysAmount < 2) {
+    void showNotEnoughKeysToast()
+    return
+  }
+
+  return keys
+}
+
 export const pickPlotConfiguration = async (): Promise<
   PlotConfigData | undefined
 > => {
-  const file = await pickDataFile()
+  const files = await pickDataFiles()
 
-  if (!file) {
+  if (!files) {
     return
   }
 
-  const data = await loadDataFile(file)
+  const fileExts = new Set(files.map(file => getFileExtension(file)))
 
-  if (!data) {
-    return Toast.showError(
-      'Failed to parse the requested file. Does the file contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
-    )
+  if (fileExts.size > 1) {
+    return Toast.showError('Files must of the same type.')
```

Comment:
Our error handling has gotten more complex with the addition of multiple files.

**Original:**
1. Check if file can be parsed.
2. Check if file holds at least two field options
3. Go on to plot template picker

**Multiple Files:**
1. Check if all files have the same extension
2. Check if all files can be parsed
3. Check if all files contain at least one field
4. Check if we have at least two fields total among all files
5. Go on to plot template picker

We could also filter out invalid files instead of failing entirely but I chose to just fail since I thought it would be less confusing. Any ideas for simplifying the logic are welcome!

Reply:
Also, all the files need to have an array of the same length but I chose not to check for that. DVC fails with this error:

<img width="527" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/ec8b275e-707d-4c8d-8c71-e504199f42aa">

We could add a check for it though if we think it would be needed.

Reply:
IMO we should be checking for things that make DVC fail and providing as much information as possible up front. Remember the main reason we're doing this is for onboarding to plots/simplification of the process.

And (as previously stated) if we fail the process because of a single file then we need to let the user know which file made the process fail and why.